tools/check-guide/src/main.rs

```diff
@@ -0,0 +1,113 @@
+    static ref TEMPLATES_PATH: PathBuf = PathBuf::from("guide/src/templates/commands/hermes/");
+
+    // List of directories which should not be visited when checking for correctness.
+    static ref EXCLUSIONS: HashSet<&'static str> = HashSet::from(["templates", "assets", "images", "theme"]);
+
+    // Regex to match macro calls in the guide
+    static ref TEMPLATE_RE: Regex = Regex::new(r"\{{2}#template.*templates/commands/hermes/.*\}\}").unwrap();
+
+    // FileReader for mdbook_template
+    static ref FILEREADER: SystemFileReader = SystemFileReader::default();
+}
+
+fn check_correctness<'a, T>(command: T) -> Result<(), ParseError>
+where
+    T: IntoIterator<Item = &'a str>,
+    // Returns an error if the command cannot be parsed.
+{
+    EntryPoint::try_parse_from(command)?;
+    Ok(())
+}
+
+fn verify_line(line: &str, path: &Path, line_number: i32) -> i32 {
+    // Extracts macro calls from a command, replace them with the content of the template and check that the command is correct.
+    // Returns the number of errors encountered.
+    let parent = path.parent().unwrap_or_else(|| Path::new(&*GUIDE_PATH));
+    let mut error_founds = 0;
+
+    TEMPLATE_RE.find_iter(line).for_each(|m| {
+        let template = m.as_str();
+        let template_replaced = replace_template(template, &*FILEREADER, &parent, "", 0);
+        if let Err(e) = check_correctness(template_replaced.split_whitespace()) {
+            eprintln!("{}:{}: {:?}", path.display(), line_number, e);
+            error_founds += 1;
+        }
+    });
+    error_founds
+}
+
+fn verify_file(path: &Path) -> i32 {
+    // Verifies that every template macro call in the file can be replaced by a valid Hermes command.
+    // Returns the number of invalid commands found.
+    let mut error_founds = 0;
+    let file = File::open(path);
+    let reader =
+        BufReader::new(file.unwrap_or_else(|_| panic!("File not found: {}", path.display())));
+    let mut line_number = 1;
+
+    for line in reader.lines() {
+        let line = line
+            .unwrap_or_else(|_| panic!("{} : Failed to read line {}", path.display(), line_number));
+        error_founds += verify_line(&line, path, line_number);
+        line_number += 1;
+    }
+    error_founds
+}
+
+fn main() {
+    // Iterate over every markdown file in the guide directory except for the excluded ones
+    let number_of_errors = WalkDir::new(GUIDE_PATH.as_path())
+        .into_iter() // Iterate over all files in the guide directory
+        .filter_entry(|e| {
+            !EXCLUSIONS.contains(
+                e.file_name()
+                    .to_str()
+                    .expect("Unwrapping a file_name to str failed."),
+            )
+        }) // Filter out the excluded directories
+        .map(|e| e.expect("Failed to get an entry."))
+        .filter(|e| e.file_type().is_file()) // Keep only files
+        .filter(|e| e.path().extension() == Some(OsStr::new("md"))) // Keep only markdown files
```

Comment:
Can these two `filter` calls be consolidated into one?
```suggestion
        .filter(|e| e.file_type().is_file() && e.path().extension() == Some(OsStr::new("md"))) // Keep only markdown files
```